internal/operator/snapshotlease/snapshotlease.go

```diff
@@ -0,0 +1,200 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package snapshotlease
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/internal/common"
+	druiderr "github.com/gardener/etcd-druid/internal/errors"
+	"github.com/gardener/etcd-druid/internal/operator/component"
+	"github.com/gardener/etcd-druid/internal/utils"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	coordinationv1 "k8s.io/api/coordination/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ErrGetSnapshotLease indicates an error in getting the snapshot lease resources.
+	ErrGetSnapshotLease druidv1alpha1.ErrorCode = "ERR_GET_SNAPSHOT_LEASE"
```

Comment:
The error code is for the component in general, and not for individual resources of the component. So a single error code for this is sufficient. The error message will have more details about the exact error encountered, which will talk about the specific resource/error. 

And since the `snapshotlease` component consists of two snapshot lease resources, `ErrGetSnapshotLease indicates an error in getting the snapshot lease resources` is the correct statement here, since this is what the error code indicates.